dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task241007CreateUniqueFieldsTable.java

```diff
@@ -0,0 +1,193 @@
+ * The contentletsId array holds the IDs of contentlets with the same field value that existed before the database was upgraded.
+ * After the upgrade, no more contentlets with duplicate values will be allowed.
+ *
+ * <b>Additional Details</b>
+ *
+ *-  The Host ID is included in the hash calculation only if the uniquePerSite field variable is enabled.
+ *- The unique_key_val field ensures that only truly unique values can be inserted moving forward.
+ *- This upgrade task also populates the unique_fields table with the existing unique field values from the current database.
+ */
+public class Task241007CreateUniqueFieldsTable implements StartupTask {
+
+    private final static String CREATE_TABLE_QUERY = "CREATE TABLE IF NOT EXISTS unique_fields (" +
+            "unique_key_val VARCHAR(64) PRIMARY KEY," +
+            "supporting_values JSONB" +
+            " )";
+
+    private static final String RETRIVE_UNIQUE_FIELD_VALUES_QUERY = "SELECT structure.inode AS content_type_id," +
+                " field.velocity_var_name AS field_var_name," +
+                " contentlet.language_id AS language_id," +
+                " identifier.host_inode AS host_id," +
+                " jsonb_extract_path_text(contentlet_as_json -> 'fields', field.velocity_var_name)::jsonb ->> 'value' AS field_value," +
+                " ARRAY_AGG(contentlet.identifier) AS contentlet_identifier" +
+            " FROM contentlet" +
+                " INNER JOIN structure ON structure.inode = contentlet.structure_inode" +
+                " INNER JOIN field ON structure.inode = field.structure_inode" +
+                " INNER JOIN identifier ON contentlet.identifier = identifier.id" +
+            " WHERE jsonb_extract_path_text(contentlet_as_json->'fields', field.velocity_var_name) IS NOT NULL" +
+            " GROUP BY structure.inode," +
+                    " field.velocity_var_name ," +
+                    " contentlet.language_id," +
+                    " identifier.host_inode," +
+                    " jsonb_extract_path_text(contentlet_as_json -> 'fields', field.velocity_var_name)::jsonb ->> 'value'";
+
+    private static final String INSERT_UNIQUE_FIELDS_QUERY = "INSERT INTO unique_fields(unique_key_val, supporting_values) VALUES(?, ?)";
+
+    @Override
+    public boolean forceRun() {
+        try {
+            final DotDatabaseMetaData databaseMetaData = new DotDatabaseMetaData();
+            return !databaseMetaData.tableExists(DbConnectionFactory.getConnection(), "unique_fields");
+        } catch (SQLException e) {
+            Logger.error(this, e.getMessage(),e);
+            return false;
+        }
+    }
+
+    @Override
+    public void executeUpgrade() throws DotDataException, DotRuntimeException {
+        createUniqueFieldTable();
+
+        try {
+            populate();
+        } catch (SQLException e) {
+            throw new DotDataException(e);
+        }
+    }
+
+    private void populate() throws DotDataException, SQLException {
```

Comment:
Missing Javadoc in all methods below

Reply:
done https://github.com/dotCMS/core/pull/30301/files#diff-93fb205d6a0d941cfc0933bbfbfc3dd5bc70723d3fc8a8513f8058939ec7f901R122